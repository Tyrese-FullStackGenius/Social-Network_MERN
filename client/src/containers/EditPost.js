import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import compose from 'recompose/compose';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({
  button: {
    margin: theme.spacing.unit
  },
  container: {
    display: 'flex',
    justifyContent: 'center'
  },
  textField: {
    marginLeft: theme.spacing.unit,
    marginRight: theme.spacing.unit,
    width: 500
  }
});

export class EditPost extends Component {
  /* eslint-disable react/destructuring-assignment */
  state = {
    postText: this.props.text
  };
  /* eslint-enable react/destructuring-assignment */

  handleChange = (e) => {
    const postText = e.target.value;
    this.setState(() => ({ postText }));
  };

  handleSubmit = (e) => {
    e.preventDefault();
    const { postText } = this.state;
    const {
      commentPostId,
      id,
      isEditingComment,
      author,
      editPost,
      handleModalClose
    } = this.props;
    if (!postText.trim()) return;

    if (isEditingComment) {
      editPost('editComment', id, commentPostId, postText);
    } else {
      editPost(id, postText, author);
    }

    handleModalClose();
  };

  render() {
    const { postText } = this.state;
    const { classes } = this.props;
    return (
      <form
        className={classes.container}
        noValidate
        autoComplete="off"
        onSubmit={this.handleSubmit}
      >
        <TextField
          id="textarea"
          placeholder="What's on your mind?"
          multiline
          className={classes.textField}
          margin="normal"
          rowsMax="5"
          value={postText}
          onChange={this.handleChange}
        />
        <Button
          variant="contained"
          color="primary"
          className={classes.button}
          type="submit"
        >
          Update
        </Button>
      </form>
    );
  }
}

EditPost.propTypes = {
  classes: PropTypes.object.isRequired,
  commentPostId: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  isEditingComment: PropTypes.bool.isRequired,
  author: PropTypes.string.isRequired,
  editPost: PropTypes.func.isRequired,
  handleModalClose: PropTypes.func.isRequired,
  text: PropTypes.string.isRequired
};

export default compose(
  withStyles(styles),
  connect()
)(EditPost);
