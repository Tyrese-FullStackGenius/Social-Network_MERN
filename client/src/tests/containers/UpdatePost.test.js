import React from 'react';
import { shallow } from 'enzyme';
import EditPost from '../../containers/EditPost';

it('should render EditPost component', () => {
  const wrapper = shallow(<EditPost />);
  expect(wrapper).toMatchSnapshot();
});
