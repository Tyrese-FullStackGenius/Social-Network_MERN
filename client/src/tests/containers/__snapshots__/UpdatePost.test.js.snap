// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditPost component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(Connect(EditPost)) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "button": "Connect-EditPost--button-1",
        "container": "Connect-EditPost--container-2",
        "textField": "Connect-EditPost--textField-3",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "button": "Connect-EditPost--button-1",
          "container": "Connect-EditPost--container-2",
          "textField": "Connect-EditPost--textField-3",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
